tests/vrf_test.py

```diff
@@ -335,6 +495,172 @@ def test_show_vnet_guid(self):
+Error: 'vnet_name' must begin with 'Vnet_' .
+"""
+        # Add the vnet to a vnet table and verify if it exists while route addition
+        args = ["Vnet_3", "2", "tunnel1"]
+        result = runner.invoke(config.config.commands["vnet"].commands["add"], args, obj=vnet_obj)
+        assert ('Vnet_3') in db.cfgdb.get_table('VNET')
+
+        # Test vnet add route using mandatory arguments
+        args = ["Vnet_3", "10.10.10.10/32", "10.10.10.1"]
+        result = runner.invoke(config.config.commands["vnet"].commands["add-route"], args, obj=vnet_obj)
+        assert any(key[0] == 'Vnet_3' for key in db.cfgdb.get_table('VNET_ROUTE_TUNNEL'))
+        assert result.exit_code == 0
+
+        # Test vnet add route using invalid vnet name
+        args = ["Vnet-3", "10.10.10.10/32", "10.10.10.1"]
+        result = runner.invoke(config.config.commands["vnet"].commands["add-route"], args, obj=vnet_obj)
+        assert result.exit_code != 0
+        assert expected_output in result.output
+
+        # Test vnet add route when vnet doesnt exist
+        args = ["Vnet_6", "10.10.10.10/32", "10.10.10.1"]
+        result = runner.invoke(config.config.commands["vnet"].commands["add-route"], args, obj=vnet_obj)
+        assert "VNET Vnet_6 doesnot exist, cannot add a route!" in result.output
+        assert result.exit_code != 0
+
+        # Test vnet add route when vnet route exists
+        args = ["Vnet_3", "10.10.10.10/32", "10.10.10.1"]
+        result = runner.invoke(config.config.commands["vnet"].commands["add-route"], args, obj=vnet_obj)
+        assert "Route already exists for the VNET Vnet_3" in result.output
+        assert result.exit_code != 0
+
+        # Test vnet add route with optional args
+        args = ["Vnet_4", "22", "tunnel1"]
+        result = runner.invoke(config.config.commands["vnet"].commands["add"], args, obj=vnet_obj)
+        assert ('Vnet_4') in db.cfgdb.get_table('VNET')
+
+        args = ["Vnet_4", "10.10.10.10/32", "10.10.10.1", "123", "8.8.8.8", "11:22:33:44:55:66"]
+        result = runner.invoke(config.config.commands["vnet"].commands["add-route"], args, obj=vnet_obj)
+        assert any(key[0] == 'Vnet_4' for key in db.cfgdb.get_table('VNET_ROUTE_TUNNEL'))
+        assert result.exit_code == 0
+
+        # Test vnet add route using length of vnet name
+        vnet_name = "Vnet_ypfbjjhyzivaythuaxlbcibgdgjkqgapedmiosjgsv"
+        args = [vnet_name, "10.10.10.10/32", "10.10.10.1"]
+        result = runner.invoke(config.config.commands["vnet"].commands["add-route"], args, obj=vnet_obj)
+        assert result.exit_code != 0
+        assert "'vnet_name' length should not exceed 15 characters" in result.output
+        assert vnet_name not in db.cfgdb.get_table('VNET_ROUTE_TUNNEL')
+
+        # Test vnet del with route deletion
+        result = runner.invoke(config.config.commands["vnet"].commands["del"], ["Vnet_4"], obj=vnet_obj)
+        assert any(key[0] != 'Vnet_4' for key in db.cfgdb.get_table('VNET_ROUTE_TUNNEL'))
```

Comment:
This should be ```all``` and not ```any```, any will only check if there is some route which exists with a key containing Vnet_4, we need to check that there are no routes containing Vnet_4.